vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/bdd/steps/integration/ResourceCheckSteps.java

```diff
@@ -121,9 +122,18 @@ private ResourceValidation validate(ResourceValidation r)
     private Stream<ResourceValidation> createResourceValidations(Stream<Element> elements,
             Function<Pair<URI, String>, ResourceValidation> resourceValidationFactory)
     {
-        return elements.map(e ->
-            Pair.of(createUri(getElementAttribute(e, "href").orElseGet(() ->
-                e.attr("src")).trim()), e.cssSelector()))
+        return elements.map(e -> {
```

Comment:
{
            Pair.of(createUri(getElementAttribute(e, "href").orElseGet(() ->	            String selector;
                e.attr("src")).trim()), e.cssSelector()))	            try
            {
                selector = e.cssSelector();
            }
            catch (SelectorParseException exception)
            {
                selector = "Unable to build selector for current element";
            }


move to separated method